control/src/Controller/CommonSMC.cpp

```diff
@@ -156,7 +156,6 @@ void CommonSMC::initialize(const Model& m)
   }
 
   _SMC->nonSmoothDynamicalSystem()->insertDynamicalSystem(_DS_SMC);
-  _SMC->nonSmoothDynamicalSystem()->topology()->setOSI(_DS_SMC, _integratorSMC);
```

Comment:
(sorry, still thinking...) On the other hand if we followed @fperignon's idea to have it call insertIntegrator(), then it would make sense for initializeOSIforDS to be placed where insertIntegrator is placed, instead of where setOSI is placed.  This is nicer, as the Simulation and OSI must already exist at that point.